Makefile

```diff
@@ -0,0 +1,139 @@
+.PHONY: test cover gofmt gofmt-fix license-check clean tar.gz docker-push release docker-push-all
+
+# Registry used for publishing images
+REGISTRY?=quay.io/coreos
+
+# Default tag and architecture. Can be overridden
+TAG?=$(shell git describe --tags --dirty)
+ARCH?=amd64
+
+# These variables can be overridden by setting an environment variable.
+TEST_PACKAGES?=pkg/ip subnet remote
+TEST_PACKAGES_EXPANDED=$(TEST_PACKAGES:%=github.com/coreos/flannel/%)
+PACKAGES?=$(TEST_PACKAGES) network
+PACKAGES_EXPANDED=$(PACKAGES:%=github.com/coreos/flannel/%)
+
+# Set the (cross) compiler to use for different architectures
+ifeq ($(ARCH),amd64)
+	CC=gcc
+	IMAGE_SUFFIX=
+endif
+ifeq ($(ARCH),arm)
+	CC=arm-linux-gnueabi-gcc
+	IMAGE_SUFFIX=-$(ARCH)
+endif
+ifeq ($(ARCH),arm64)
+	CC=aarch64-linux-gnu-gcc
+	IMAGE_SUFFIX=-$(ARCH)
+endif
+ifeq ($(ARCH),ppc64le)
+	CC=powerpc64le-linux-gnu-gcc
+	IMAGE_SUFFIX=-$(ARCH)
+endif
+GOARM=6
+KUBE_CROSS_TAG=v1.6.2-2
+
+dist/flanneld: $(shell find . -type f  -name '*.go')
+	go build -o dist/flanneld \
+	  -ldflags "-extldflags -static -X github.com/coreos/flannel/version.Version=$(TAG)"
+
+test: license-check gofmt
```

Comment:
The tests always run with coverage. The `cover` target is just generating the html report, but it can only do that for a single package. Hence they are kept separate.